typespec-extension/src/type-utils.ts

```diff
@@ -51,6 +52,24 @@ export class ProcessingCache<In, Out> {
   }
 }
 
+/**
+ * Expose `isStable` and comparison capability of an Api Version.
+ */
+export class ComparableApiVersion {
+  version: Version;
+  constructor(version: Version) {
+    this.version = version;
+  }
+
+  isStable(): boolean {
+    return !this.version.value.includes("preview");
+  }
+
+  noLaterThan(anotherVersion: ComparableApiVersion): boolean {
+    return this.version.value <= anotherVersion.version.value;
```

Comment:
which is larger? `2022-02-01` or `2022-02-01-preview`

I don't have idea though.

Reply:
Emmm, yes, former should be larger.. I think I should borrow the logic from autocent.js..

Reply:
Or maybe we should rely on service to do the sort for us. We only filter out those appear after the pinned version, in case of some strange `2022-12-01-preview2` or `2022-12-01-newpreview`.